{"additions": 38, "auther_ref": "unitTestScriptBug", "auther_sha": "44f7106bc6843bf27930dab81a5a36d2056565fa", "author": "wddgit", "body": "#### PR description:\r\n\r\nFix bugs in several of the Framework unit test scripts. These affected places where a unit test ran an executable that was expected and required to fail. Most of the time it was a cmsRun executable.\r\n\r\nIf the executable actually failed as intended, then no problem would occur. The unit test would behave as intended. In all but two of the cases this was happening. The only fix needed is to change \"$?\" to \"1\" in one place in the bash script. I searched for these and fixed them all.\r\n\r\nIf an executable that was supposed to fail didn't fail, the problem would not be reported as a unit test failure and the script would stop running at that point.\r\n\r\nCatchCmsExceptiontest.sh and CatchStdExceptiontest.sh were the two other cases. The problems there originated in December 2007 (surprisingly none of us ever noticed, first time I've looked into CVS in a  long time, the CVS browser still works).  The configurations that were running actually don't fail, but the problem is in the test configurations. I fixed them so they fail in what I've guessed is the originally intended manner. I also updated the BuildFile to run these two scripts in the new way.\r\n\r\nIn CatchCmsExceptiontest.sh, the script stopped and the unit tests after the first were not being run. Now that this is fixed, two of them are failing: testSkipEvent_cfg.py and CatchCmsExceptionFromSource_cfg.py. In this PR, I just commented these out. My intent is to create an issue for each of these for someone to investigate and resolve at some point in the future and fix in a different PR. It is possible these tests are simply obsolete. The skipEvent test is related to the behavior after an exception. I'm not sure how this is supposed to work with concurrency changes that have occurred since the test was created. I didn't investigate the other one at all. These problems are somewhat independent of the fixes in this PR. They were not running before because of the bug, they won't run after the PR because I commented them out.\r\n\r\n#### PR validation:\r\n\r\nThe  fixed unit tests pass. Nothing was changed that could affect anything other than unit tests. Our test coverage was not as good as we thought, but there is no evidence any of any problem that could affect production/analysis job output or behavior.\r\n", "branch": "master", "changed_files": 15, "closed_at": "1625579400", "comments": 7, "commits": 1, "created_at": "1625254028", "deletions": 34, "labels": ["code-checks-approved", "core-approved", "fully-signed", "orp-approved", "tests-approved"], "merge_commit_sha": "3758d61efb4b4374fc9b06b22a33985888edc959", "merged_at": "1625579400", "merged_by": "cmsbuild", "milestone": "CMSSW_12_0_X", "number": 34321, "release-notes": [], "review_comments": 0, "state": "closed", "title": "Fix Framework unit test script bugs", "updated_at": "1625579400", "user": "wddgit"}